src/features/completionProvider.ts

```diff
@@ -0,0 +1,93 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { CompletionItemProvider, TextDocument, Position, CompletionContext, CompletionList, CompletionItem, MarkdownString, TextEdit, Range, SnippetString } from "vscode";
+import AbstractProvider from "./abstractProvider";
+import * as protocol from "../omnisharp/protocol";
+import * as serverUtils from '../omnisharp/utils';
+import { CancellationToken, CompletionTriggerKind as LspCompletionTriggerKind, InsertTextFormat } from "vscode-languageserver-protocol";
+import { createRequest } from "../omnisharp/typeConversion";
+
+export default class OmnisharpCompletionProvider extends AbstractProvider implements CompletionItemProvider {
+
+    #lastCompletions?: Map<CompletionItem, protocol.OmnisharpCompletionItem>;
```

Comment:
Oh snap. Is this our first private field?

Reply:
I have no idea, I search "typescript private field" and found this :smile:.